knowledge_graph_class.py

```diff
@@ -0,0 +1,94 @@
+"""Knowledge Graph"""
+from typing import Dict, List, Any
+from transformers import (
+    Text2TextGenerationPipeline,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer)
+from transformers.pipelines.text2text_generation import ReturnType
+from pydantic import Field
+
+class KnowledgeGraph():
+  """Knowledge Graph class"""
+
+  max_new_tokens: int = Field(
+    default=200, description="Maximum number of new tokens to generate."
+  )
+  predictions : List[Dict[str, Any]] = Field(default=[])
+
+  def __init__(
+    self,
+    *args,
+    model_name:str='Babelscape/rebel-large',
+    **kwargs):
+
+    self.model = EntityRelationshipExtractionPipeline(model=model_name, *args, **kwargs)
+
+  def inference(self, sentence: str) -> Dict[str, Any]:
+    """inference"""
+    self.predictions = self.model(sentence)
+
+    return self.predictions
+
+class EntityRelationshipExtractionPipeline(Text2TextGenerationPipeline):
+  """Key phrase extractor pipeline class"""
+  def __init__(self, model, *args, **kwargs):
+    super().__init__(
+      model=AutoModelForSeq2SeqLM.from_pretrained(model),
+      tokenizer=AutoTokenizer.from_pretrained(model),
+      *args,
+      **kwargs)
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `EntityRelationshipExtractionPipeline` class, you're using `AutoModelForSeq2SeqLM.from_pretrained(model)` and `AutoTokenizer.from_pretrained(model)`. This assumes that the model and tokenizer are always available and compatible. It would be better to handle potential exceptions here for robustness.

```diff
-  model=AutoModelForSeq2SeqLM.from_pretrained(model),
-  tokenizer=AutoTokenizer.from_pretrained(model),

+  try:
+    model=AutoModelForSeq2SeqLM.from_pretrained(model),
+    tokenizer=AutoTokenizer.from_pretrained(model),
+  except Exception as e:
+    raise ValueError(f"Failed to load model or tokenizer with name {model}: {e}")
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->